### PR TITLE
Adding support for thumbnail and constant image limit (10)

### PR DESCRIPTION
**Pull request by s-varun**

### CODE REVIEW

File: client/constants.py [hunk 1]
```diff
@@ -1,3 +1,9 @@
+class ImageAttrs(object):
+    IMAGE_HEIGHT = 50
+    IMAGE_WIDTH = 50
+    IMAGES_NUMBER = 10
```
Comment (MeenuV) [inline client/constants.py]: Rename to something like `MAX_RETURN_IMAGES` (along those lines)

File: client/scraper/metadata.py [hunk 2]
```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
...
         if len(image_urls) == 0:
             image_urls = soup.findAll('img')
             image_attr = 'src'
-            prepend_url = self.prop_map[FieldKeyword.PROVIDER_URL]
             if len(image_urls) == 0:
                 return None
-        images_list[FieldKeyword.COUNT] = 0
-        images_list[FieldKeyword.DATA] = []
+
         for i in range(len(image_urls)):
```
Comment (MeenuV) [inline client/scraper/metadata.py]: Why not just say `for image_url in image_urls:` ?

File: client/scraper/metadata.py [hunk 3]
```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
...
+            if image.has_attr(image_attr):
+                image_url = image[image_attr]
+                prepend_url = self._get_prepend_url(image_url)
+                if not (image.has_attr('height') and int(image['height']) < ImageAttrs.MIN_IMAGE_HEIGHT):
+                    if image_url is "":
+                        continue
+                    image_item_dict[FieldKeyword.URL] = (prepend_url + image_url).encode('utf-8')
+                    images_list[FieldKeyword.DATA].append(image_item_dict)
+                    images_list[FieldKeyword.COUNT] = images_list[FieldKeyword.COUNT] + 1
+                    if images_list[FieldKeyword.COUNT] >= ImageAttrs.MAX_RETURN_IMAGES:
```
Comment (AliceYuan) [inline client/scraper/metadata.py]: I think `==` is fine?

File: client/scraper/metadata.py [hunk 4]
```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
...
-                image_item_dict[FieldKeyword.URL] = (prepend_url + image_urls[i][image_attr]).encode('utf-8')
-                images_list[FieldKeyword.DATA].append(image_item_dict)
-                images_list[FieldKeyword.COUNT] = images_list[FieldKeyword.COUNT] + 1
+            if image.has_attr(image_attr):
+                image_url = image[image_attr]
+                prepend_url = self._get_prepend_url(image_url)
+                if not (image.has_attr('height') and int(image['height']) < ImageAttrs.MIN_IMAGE_HEIGHT):
+                    if image_url is "":
+                        continue
+                    image_item_dict[FieldKeyword.URL] = (prepend_url + image_url).encode('utf-8')
```
Comment (AliceYuan) [inline client/scraper/metadata.py]: there's a url function in `url_utils.py` called `validate_image_url`. Can you call that? it makes sure all the images that are relative urls are fixed.

Comment (AliceYuan) [inline client/scraper/metadata.py]: also, move the encode into `validate_image_url`

File: client/scraper/metadata.py [hunk 5]
```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
...
+
     def get_images_list(self, response):
         soup = BeautifulSoup(response.content)
         images_list = collections.OrderedDict()
-        image_urls = soup.findAll(MetadataFields.META, attrs={MetadataFields.PROPERTY: MetadataFields.OG_IMAGE})
+        images_list[FieldKeyword.COUNT] = 0
+        images_list[FieldKeyword.DATA] = []
+
+        images = soup.findAll(MetadataFields.META, attrs={MetadataFields.PROPERTY: MetadataFields.OG_IMAGE})
         image_attr = 'content'
```
Comment (AliceYuan) [inline client/scraper/metadata.py]: c

File: client/scraper/metadata.py [hunk 6]
```diff
@@ -106,26 +108,43 @@ def get_desc(self, response):
...
-        prepend_url = ''
-        if len(image_urls) == 0:
-            image_urls = soup.findAll('img')
+        if len(images) == 0:
+            images = soup.findAll('img')
             image_attr = 'src'
```
Comment (AliceYuan) [inline client/scraper/metadata.py]: change this to be a constant? `MetadataFields.SRC`

**Comment by AliceYuan** [comment]
r+ after changes are made

**Pull request merged**